### PR TITLE
feat(api): introduce BrowserContext.waitForEvent

### PR DESCRIPTION
**Pull request by yury-s**

### CODE REVIEW

File: src/browserContext.ts [hunk 1]
```diff
@@ -50,15 +52,69 @@ export interface BrowserContext {
...
+
+  setDefaultNavigationTimeout(timeout: number) {
+    this._timeoutSettings.setDefaultNavigationTimeout(timeout);
+  }
+
+  setDefaultTimeout(timeout: number) {
+    this._timeoutSettings.setDefaultTimeout(timeout);
+  }
+
+  async waitForEvent(event: string, optionsOrPredicate?: Function | (types.TimeoutOptions & { predicate?: Function })): Promise<any> {
```
Comment (dgozman) [inline src/browserContext.ts]: Let's also move `_browserClosed`.

File: src/firefox/ffPage.ts [hunk 2]
```diff
@@ -45,7 +45,7 @@ export class FFPage implements PageDelegate {
   private _eventListeners: RegisteredListener[];
   private _workers = new Map<string, { frameId: string, session: FFSession }>();
 
-  constructor(session: FFSession, browserContext: BrowserContext, openerResolver: () => Promise<Page | null>) {
+  constructor(session: FFSession, browserContext: BrowserContextBase, openerResolver: () => Promise<Page | null>) {
```
Comment (dgozman) [inline src/firefox/ffPage.ts]: Let's use FFBrowserContext.

File: test/browsercontext.spec.js [hunk 3]
```diff
@@ -389,7 +396,7 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, FF
       const context = await browser.newContext();
       const page = await context.newPage();
       const [otherPage] = await Promise.all([
-        new Promise(r => context.once('page', async event => r(await event.page()))),
+        context.waitForEvent('page').then(async event => await event.page()),
```
Comment (dgozman) [inline test/browsercontext.spec.js]: nit: arrow function can we written as `...then(e => e.page())`

File: test/chromium/launcher.spec.js [hunk 4]
```diff
@@ -83,7 +83,7 @@ module.exports.describe = function({testRunner, expect, defaultBrowserOptions, p
       const backgroundPages = await context.backgroundPages();
       let backgroundPage = backgroundPages.length
           ? backgroundPages[0]
-          : await new Promise(fulfill => context.once('backgroundpage', async event => fulfill(await event.page())));
+          : await context.waitForEvent('backgroundpage').then(async event => await event.page());
```
Comment (dgozman) [inline test/chromium/launcher.spec.js]: ditto

File: test/popup.spec.js [hunk 5]
```diff
@@ -29,7 +29,7 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
       await page.setContent('<a target=_blank rel=noopener href="/popup/popup.html">link</a>');
       const requestPromise = server.waitForRequest('/popup/popup.html');
       const [popup] = await Promise.all([
-        new Promise(fulfill => context.once('page', async pageEvent => fulfill(await pageEvent.page()))),
+        context.waitForEvent('page').then(async pageEvent => await pageEvent.page()),
```
Comment (dgozman) [inline test/popup.spec.js]: ditto

File: src/browserContext.ts (hunk 1 above)
Comment (yury-s) [reply src/browserContext.ts]: Done.

File: src/firefox/ffPage.ts (hunk 2 above)
Comment (yury-s) [reply src/firefox/ffPage.ts]: Done.

File: test/browsercontext.spec.js (hunk 3 above)
Comment (yury-s) [reply test/browsercontext.spec.js]: Done.

File: test/chromium/launcher.spec.js (hunk 4 above)
Comment (yury-s) [reply test/chromium/launcher.spec.js]: Done.

File: test/popup.spec.js (hunk 5 above)
Comment (yury-s) [reply test/popup.spec.js]: Done.

**Review by dgozman — APPROVED** [review]

**Pull request merged**